test/IntegrationTests/AspNetTests.cs

```diff
@@ -68,15 +68,13 @@ public async Task TracesResource()
     {
         Assert.True(EnvironmentTools.IsWindowsAdministrator(), "This test requires Windows Administrator privileges.");
 
-        _environmentVariables["OTEL_SERVICE_NAME"] = ServiceName;
-
         // Using "*" as host requires Administrator. This is needed to make the mock collector endpoint
         // accessible to the Windows docker container where the test application is executed by binding
         // the endpoint to all network interfaces. In order to do that it is necessary to open the port
         // on the firewall.
         using var collector = new MockSpansCollector(Output, host: "*");
         using var fwPort = FirewallHelper.OpenWinPort(collector.Port, Output);
-        collector.ResourceExpector.Expect("service.name", ServiceName); // this is set via env var in Dockerfile and Wep.config, but env var has precedence
+        collector.ResourceExpector.Expect("service.name", "Default Web Site"); // this is set by the ServiceNameDetector
```

Comment:
I think this should be reverted. The old code has `OTEL_SERVICE_NAME` set to specific value, this value should be present. `service.name` detector tests should be separate.

Reply:
I will make a new Integration Test for this then

Reply:
We are OK with a whole new Int Test for this change?

Reply:
As far if the unit test can verify the priority also, I'm fine.